python/src/aac/plugins/gen_design_doc/gen_design_doc_impl.py

```diff
@@ -97,8 +97,13 @@ def _get_and_prepare_definitions_by_type(parsed_definitions: list[Definition], a
     return [definition.structure for definition in filtered_definitions]
```

Comment:
I'm not able to highlight-comment the function `_get_and_prepare_definitions_by_type`, but this solution would resolve the need to have default values in the `Definition` class and the function `_get_interfaces_with_required_fields`.

```python
def _get_and_prepare_definitions_by_type(parsed_definitions: list[Definition], aac_type: str) -> list[dict]:
    def get_definition_structure_with_required_fields(interface_definition: Definition):
        return interface_definition.structure | {"required_fields": get_required_fields(interface_definition)}

    filtered_definitions = get_definitions_by_root_key(aac_type, parsed_definitions)
    definition_template_properties = []
    if aac_type == "schema":
        definition_template_properties = [get_definition_structure_with_required_fields(definition) for definition in filtered_definitions]
    else:
        definition_template_properties = [definition.structure for definition in filtered_definitions]

    return definition_template_properties
```

Reply:
@cvchaparro , do you want me to go ahead and implement this? What are you thoughts

Reply:
Pushed this change through (See recent comment)